addons/hr_holidays/models/hr_leave.py

```diff
@@ -296,155 +296,153 @@ def _search_description(self, operator, value):
         leaves = self.search(domain)
         return [('id', 'in', leaves.ids)]
 
-    @api.onchange('holiday_status_id')
-    def _onchange_holiday_status_id(self):
-        self.request_unit_half = False
-        self.request_unit_hours = False
-        self.request_unit_custom = False
-        self.state = 'confirm' if self.validation_type != 'no_validation' else 'draft'
-
-    @api.onchange('request_date_from_period', 'request_hour_from', 'request_hour_to',
-                  'request_date_from', 'request_date_to',
-                  'employee_id')
-    def _onchange_request_parameters(self):
-        if not self.request_date_from:
-            self.date_from = False
-            return
+    @api.depends('holiday_status_id')
+    def _compute_state(self):
+        for holiday in self:
+            holiday.state = 'confirm' if holiday.validation_type != 'no_validation' else 'draft'
 
-        if self.request_unit_half or self.request_unit_hours:
-            self.request_date_to = self.request_date_from
+    @api.depends('request_date_from_period', 'request_hour_from', 'request_hour_to', 'request_date_from', 'request_date_to',
+                'request_unit_half', 'request_unit_hours', 'request_unit_custom', 'employee_id')
+    def _compute_date_from_to(self):
+        for holiday in self:
+            if not holiday.request_date_from:
+                holiday.date_from = False
+                return
```

Comment:
Do not return inside the for loop! Otherwise the fields won't be assigned for any following records in self.

Reply:
FYI, `continue` would have worked too ;).  But `if elif else` is fine ;).
```python
def one_record(self):
  if x:
    return
  ...

def multi_records(self):
  for record in self:
    if x:
      continue
    ...
```